File: solutions/beecrowd/2460/2460.py
import sys

for line in sys.stdin:
    people_in_line = list(map(int, input().split()))
    _ = input()
    people_who_left_line = set(map(int, input().split()))
    people_that_remained_in_line = set(people_in_line) - people_who_left_line

    first_person = True
    for person in people_in_line:
        if person in people_that_remained_in_line:
            if not first_person:
                print(' ', end='')
            print(person, end='')
            first_person = False
    print()

File: solutions/beecrowd/2460/2460.cpp
#include <cstdint>
#include <iostream>
#include <set>
#include <vector>

int main() {
    std::ios_base::sync_with_stdio(false);
    std::cin.tie(0);

    int32_t n, m, person;
    bool first_person;
    std::vector< int32_t > people_in_line;
    std::set< int32_t > people_who_left_line;

    while (std::cin >> n) {
        for (int32_t i = 0; i < n; i++) {
            std::cin >> person;
            people_in_line.push_back(person);
        }

        std::cin >> m;
        for (int32_t i = 0; i < m; i++) {
            std::cin >> person;
            people_who_left_line.insert(person);
        }

        first_person = true;
        for (int32_t person : people_in_line) {
            if (people_who_left_line.contains(person)) {
                continue;
            }

            if (!first_person) {
                std::cout << ' ';
            }

            std::cout << person;
            first_person = false;
        }

        std::cout << '\n';

        people_in_line.clear();
        people_who_left_line.clear();
    }

    return 0;
}
